### DISCUSSION TITLE
Access to metadata of the underlying model

### DISCUSSION THREAD

**Discussion by shadowfaxproject** [abetlen/llama-cpp-python#953] [Q&A]
Is there a way to access (print) metadata such as `general.name` or `model type` from the python code when using llama-cpp-python. I am able to access the standard params provided by the user but not the core metadata of the model.

**Comment by rrrusst** [comment]
I'd be interested to know if this is possible as well. That way, we can have a program that changes our preset prompt formats based on the type of model, since some models use <instruction> whereas others use [INST].

**Reply by abetlen** [reply]
@shadowfaxproject @rrrusst yes, recently added, you can now access the `metadata` property of the `Llama` class, it's all stored there in a dictionary.
